StitchCore/StitchCore/Source/Auth/AuthProviderInfo.swift

```diff
@@ -1,14 +1,101 @@
 import Foundation
+import ExtendedJson
 
-/// Struct containing information about available providers
-public struct AuthProviderInfo: Codable {
-    private enum CodingKeys: String, CodingKey {
-        case anonymousAuthProviderInfo = "anon/user"
-        case googleProviderInfo = "oauth2/google"
-        case facebookProviderInfo = "oauth2/facebook"
-        case emailPasswordAuthProviderInfo = "local/userpass"
+public protocol AuthProviderType: Codable {
+    var id: String { get }
+    var name: String { get }
+    var type: String { get }
+    var disabled: Bool { get }
```

Comment:
[whoops] I'm going to make this not show to clients. You should only ever see enabled providers anyway. You will only get name, type, metadata_fields, and config